File: java-apps/kq-common/src/main/java/com/kafkaquest/kq/common/util/models/events/eventdata/PieceAddedEventData.java
package com.kafkaquest.kq.common.util.models.events.eventdata;

import com.kafkaquest.kq.common.util.models.events.other.TransportScoreCard;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@EqualsAndHashCode(callSuper = false)
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PieceAddedEventData extends EventData {

    private UUID pieceId;
    private int movedRowPosition;
    private int movedColumnPosition;
    private Boolean hasMovedToCorrectPosition;
    private TransportScoreCard scoreCard;

}

File: kq-projection-dotnet/kq-projection/Common/Models/Events/PieceAddedEventData.cs
namespace KQ_Projection.Common.Models.Events;

using KQ_Projection.Common.Models.Jigsaw;
using System.Text.Json.Serialization;

public class PieceAddedEventData : EventData
{
    [JsonPropertyName("pieceId")]
    public Guid pieceId { get; set; }


    [JsonPropertyName("movedRowPosition")]
    public int movedRowPosition { get; set; }


    [JsonPropertyName("movedColumnPosition")]
    public int movedColumnPosition { get; set; }

    [JsonPropertyName("hasMovedToCorrectPosition")]
    public bool hasMovedToCorrectPosition { get; set; }

    [JsonPropertyName("scoreCard")]
    public TransportScoreCard? scoreCard { get; set; }
}
